src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -842,12 +843,10 @@ private void WriteAnalyzers()
 
                         if (targetLibraries == null)
                         {
-                            targetLibraries = _runtimeTarget
-                                .Libraries
-                                .ToDictionary(l => l.Name, StringComparer.OrdinalIgnoreCase);
```

Comment:
Is it a problem to lose the case-insensitivity?

Reply:
I was wondering the same thing.  I think that it's probably safe to assume that package IDs use consistent casing in the assets file.  @nkolev92 @zkat Does that sound correct?

Reply:
It really depends.
Some sections of the assets file represent user input and those honor the user casing, targets/libraries sections should be ok, but I don't think we have any code that assumes that.

So it really depends on what you are comparing.

Reply:
@nkolev92 In this case we're comparing `LockFileTargetLibrary` and `LockFileLibrary`.

Reply:
I think it should be safe then. 

You can check that easily by testing with `newtonsoft.json` or `nuget.common`.